### DISCUSSION TITLE
Do watch()/useWatch() clone objects?

### DISCUSSION THREAD

**Discussion by mordechaim** [react-hook-form/react-hook-form#4168] [Q&A] [answered]
Hi,

I found in my project that watching an array of strings with `watch()` or `useWatch()` returns a cloned array each time while `getValues()` doesn't. Is this how it's supposed to work or is it a bug somewhere in my code?

If it's intended, how can I avoid it but still trigger re-renders when the value changes (i.e. I can't use `getValues()`)?

Thanks in advanced

**Comment by bluebill1049** [comment]
we don't clone return value. you want to share an csb with issue?

**Reply by mordechaim** [reply]
It's part of a large project and hard to slim down. So it means the issue is on my end.

**Reply by mordechaim** [reply]
I'll try to share a csb if I can reproduce it in standalone

**Reply by bluebill1049** [reply]
yea, sure the issue. let's have a look at it.

**Reply by mordechaim** [reply]
Clearly not cloning in a minimalistic cbs. Thanks for you assistance

**Reply by bluebill1049** [reply]
no worries.

**Comment by mordechaim** [comment, marked as answer]
I successfully reproduced this issue in [this CSB](https://codesandbox.io/s/fervent-heyrovsky-fc9xy?file=/src/App.js)

What I noticed was that setting `shouldUnregister: false` causes this behavior. Commenting out that line will resolve it.

**Reply by bluebill1049** [reply]
oh, that's right... `shouldUnregister: false`, do a clone of the input values, so when inputs get unmounted, the value is retained. I think that's a design limitation if you encounter this issue, we are fixing that in V7.

**Reply by mordechaim** [reply]
Using `useController().value` will work. It rerenders on changes and won't clone.

**Reply by bluebill1049** [reply]
I see. that's a good solution, we will no longer clone anything in v7. FYI: we will no longer running into this sort of issue.
